File: 프로그래머스/lv2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = 1000
    st=0
    en=0
    l=0
    while en-st+1<=len(s): 
        cnt=1
        l+=1
        st=0
        en=l-1
        temp=s[st:en+1]
        cur_letter=""
        st+=l
        en+=l
        while True:
            if temp!=s[st:en+1]:
                if cnt==1:
                    cur_letter+=temp
                else:
                    cur_letter+=str(cnt)+temp
                temp=s[st:en+1]
                cnt=1
            else:
                cnt+=1
            if en>=len(s):
                cur_letter+=temp
                break
            st+=l
            en+=l
        answer=min(answer,len(cur_letter))
    return answer

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
class Solution {

    private int maxLen;
    
    public int solution(String s) {
        maxLen = s.length();
        int answer = maxLen;
        
        for(int len=1; len<=maxLen; len++){
            answer = Math.min(answer, calculateLen(s, len));
        }
        
        return answer;
    }
    
    public int calculateLen(String s, int len){
        int total = 0;
        Integer count = 0;
        String prev = s.substring(0, len);
        for(int i=0; i<maxLen; i+=len){
            String cur = s.substring(i, Math.min(i+len, maxLen));
            if(prev.equals(cur)){
                count ++;
            }else{
                prev = cur;
                total += len;
                if(count>1){
                    total += count.toString().length();
                }
                count = 1;
            }
        }
        
        if(count>1){
            total += count.toString().length();
        }
        total += prev.length();
        return total;
    }
}